### ISSUE TITLE
Expose via Web API

### ISSUE THREAD

**Issue by lqdev** [microsoft/markitdown#133]
Set up a Web API so users can use the library via a REST endpoint. 

This is also useful for Docker scenarios as well.

**Comment by lqdev** [comment]
Feel free to assign this one to me.

**Comment by gautam-e** [comment]
I started working on something like this but the moment I import `from markitdown import MarkItDown` I get the ugly warning:
```
python3.12/site-packages/starlette/routing.py:297: ResourceWarning: Unclosed file <tempfile.SpooledTemporaryFile object at 0x149021990>
  await self.app(scope, receive, send)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
```
This is without even actually instantiating a MarkItDown object let alone using it. Is that a bug in the package?

**Comment by gautam-e** [comment]
Here's a web app I made that works with Markitdown. I use it for my personal workflow: www.docx2md.com

**Comment by luisquintanilla** [comment]
Great to see the community work and samples around this! 

@elbruno also put together a quick sample of how such an API might be consumed in a C# client application. 

https://github.com/elbruno/MarkItDownServer

Once #202 is merged, it'd be great to publish the Dockerfile as an image to the Microsoft Container Registry so you don't have to pull down the entire repo to use the container / server.

**Comment by dezoito** [comment]
Any updates on this?

**Comment by afourney** [comment]
Thanks for the suggestion. This repository is intended to provide Python libraries that can be incorporated into other systems, and additional web servers or REST APIs are now explicitly out of scope. A separate package or project depending on markitdown would be the appropriate home for this. I’m closing this as not planned.
